dist/tools/ci/print_toolchain_versions.sh

```diff
@@ -5,21 +5,17 @@ get_cmd_version() {
         return
     fi
 
-    local cmd="$1"
-    if command -v "$cmd" 2>&1 >/dev/null; then
-        ver=$("$cmd" --version 2> /dev/null | head -n 1)
-        # some tools (eg. openocd) print version info to stderr
-        if [ -z "$ver" ]; then
-            ver=$("$cmd" --version 2>&1 | head -n 1)
-        fi
-        if [ -z "$ver" ]; then
-            ver="error"
-        fi
-    else
-        ver="missing"
+    VERSION_RAW=$( ($@ --version) 2>&1)
```

Comment:
changed behaviour, was `2> /dev/null` which is actually needed because with `2>&1` you get the following on macOS:

```
python3 -m flake8 --version 2>&1
/usr/local/lib/python3.7/site-packages/pycodestyle.py:113: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
3.5.0 (mccabe: 0.6.1, pycodestyle: 2.3.1, pyflakes: 1.6.0) CPython 3.7.0 on Darwin
```

while as before:

```
python3 -m flake8 --version 2>/dev/null
3.5.0 (mccabe: 0.6.1, pycodestyle: 2.3.1, pyflakes: 1.6.0) CPython 3.7.0 on Darwin
```

Reply:
That is because your version of `pycodestyle` is old. I can reproduce the issue myself if I force the installation of those versions via PIP.

In any case, if I change the behaviour I'd fix it, but I'd rather handle that case by case, see my comment below.